toolchain/check/dump_id.h

```diff
@@ -0,0 +1,51 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_DUMP_ID_H_
+#define CARBON_TOOLCHAIN_CHECK_DUMP_ID_H_
+
+#include "toolchain/lex/token_kind.h"
+#include "toolchain/lex/tokenized_buffer.h"
+#include "toolchain/parse/tree.h"
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::Check {
+
+class Context;
+
+namespace DumpIdOverloads {
```

Comment:
Would it work to omit the namespace and instead give these overloads some name that's unlikely to collide, like `DumpIdImpl`? If so, I think I'd prefer that; namespace nesting can have surprising effects (see e.g. [here](https://abseil.io/tips/130)), so I find it slightly clearer to use name affixes instead, where practical.

Reply:
Yep, done